### ISSUE TITLE
Demangler is too aggressive

### ISSUE THREAD

**Issue by Maykeye** [Vector35/binaryninja-api#381]
Let's say we have a.c

```
int a(int b, int c) { return b+c; } 
int main() { return a(3,4); }
```

compiled like `gcc -m32 a.c`  (with  gcc (Ubuntu 5.4.0-6ubuntu1~16.04.2) 5.4.0 20160609)

Demo version  thinks that function `a` is actually called `signed char` (which is what g++ us for mangling signed char)

http://imgur.com/a/5Devk

**Comment by psifertex** [comment]
We're just about done with a totally new demangler, we'll make sure this test case works in it. Thanks for the report! (As an aside, right now we're just using the OS native demangling so that's odd that it's wrong)

**Comment by plafosse** [comment]
The problem is that we aren't checking for "__Z" before passing it to the demangler, this will be fixed when the new demangler is integrated

**Comment by plafosse** [comment]
Will be fixed when #18 is done.

**Comment by plafosse** [comment]
Fixed in Stable
